### ISSUE TITLE
Using an undefined variable in try branch leads to a blocking error

### ISSUE THREAD

**Issue by xiaxinmeng** [python/mypy#15391]
'try-except' structure is used to  handle errors in a program. Errors include the NameError, e.g., using an undefined variable. As a result, the following code should work fine, while mypy reports a blocking error.

mypy_example.py
```
try: 
    TYPE_CHECKING  
except: 
    TYPE_CHECKING = False  
```

### The expected behavior
should work fine here.

### The actual output:
```
>> mypy 'mypy_example.py' 
mypy_example.py:28: error: invalid syntax  [syntax]
Found 1 error in 1 file (errors prevented further checking)
```

### Test Environment
```
mypy 1.4.0+dev
Python 3.11.3
Ubuntu 18.04

```

**Comment by JelleZijlstra** [comment]
I can't reproduce this: https://mypy-play.net/?mypy=latest&python=3.10&gist=d57720e8c2e116169ebe7d695efb7e01 just produces a `used-before-def` error as expected.

**Comment by xiaxinmeng** [comment]
Yes, it was a mistake on my part. It does not generate a blocking error for this case. The following one reports a blocking error, but i think it is not a bug.

```
try: 
    TYPE_CHECKING  #type: bool
except: 
    TYPE_CHECKING = False 
```
